[libpas] Remove partial views

Achierius opened the pull request:
#### 99779db9ef0c3057b52e8d2d233728ddbb55d3ae
<pre>
[libpas] Remove partial views
<a href="https://bugs.webkit.org/show_bug.cgi?id=304655">https://bugs.webkit.org/show_bug.cgi?id=304655</a>
<a href="https://rdar.apple.com/151419388">rdar://151419388</a>

Reviewed by Yusuke Suzuki.

Partial views have been completely disabled for some time, as the
benefits they gave (&lt;1% memory savings) were not worth the complexity,
for reasons of security, performance, and maintanance burden.
In particular, the performance-memory tradeoff was unfavorable, and we
were able to pay for their disablement with some changes to the
object-size-class allocation layout.
As this code is unused, it should be removed to reduce complexity and
improve speed of development.  This patch tears out the implementation
for partial views, including
 - partial views
 - shared views
 - shared directories
and other affiliated structures.

* Source/bmalloc/libpas/src/libpas/jit_heap_config.c:
(jit_small_segregated_shared_page_directory_selector): Deleted.
(jit_heap_config_for_each_shared_page_directory): Deleted.
(jit_heap_config_for_each_shared_page_directory_remote): Deleted.
(jit_heap_config_dump_shared_page_directory_arg): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_all_heaps.c:
(for_each_segregated_directory_shared_page_directory_callback): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_all_shared_page_directories.c: Removed.
* Source/bmalloc/libpas/src/libpas/pas_all_shared_page_directories.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_basic_heap_page_caches.h:
(pas_basic_heap_page_caches_get_shared_page_directories): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_compact_atomic_segregated_partial_view_ptr.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_compact_segregated_shared_page_directory_ptr.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_compact_segregated_shared_view_ptr.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_enumerate_segregated_heaps.c:
(collect_shared_page_directories_shared_page_directory_callback): Deleted.
(collect_shared_page_directories_heap_callback): Deleted.
(enumerate_shared_view): Deleted.
(enumerate_partial_view): Deleted.
(shared_page_directory_view_callback): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_full_alloc_bits_inlines.h:
(pas_full_alloc_bits_create_for_view_and_directory):
(pas_full_alloc_bits_create_for_view):
(pas_full_alloc_bits_create_for_partial_but_not_primordial): Deleted.
(pas_full_alloc_bits_create_for_partial): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_heap_config_utils.c:
(pas_heap_config_utils_for_each_shared_page_directory): Deleted.
(pas_heap_config_utils_for_each_shared_page_directory_remote): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_local_allocator_config_kind.h:
(pas_local_allocator_config_kind_is_primordial_partial): Deleted.
(pas_local_allocator_config_kind_create_primordial_partial): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_local_allocator_inlines.h:
(pas_local_allocator_set_up_primordial_bump): Deleted.
(pas_local_allocator_start_allocating_in_primordial_partial_view): Deleted.
(pas_local_allocator_bless_primordial_partial_view_before_stopping): Deleted.
(pas_local_allocator_try_allocate_in_primordial_partial_view): Deleted.
(pas_local_allocator_return_memory_to_page_for_role): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_page_kind.h:
(pas_page_kind_get_segregated_role): Deleted.
(pas_page_kind_for_segregated_variant_and_role): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_segregated_page_config.h:
(pas_segregated_page_config_payload_offset_for_role): Deleted.
(pas_segregated_page_config_payload_size_for_role): Deleted.
(pas_segregated_page_config_payload_end_offset_for_role): Deleted.
(pas_segregated_page_config_enable_empty_word_eligibility_optimization_for_role): Deleted.
(pas_segregated_page_config_logging_mode_for_role): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_segregated_page_config_kind_and_role.c: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_page_config_kind_and_role.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_page_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_page_role.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_partial_view.c: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_partial_view.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_partial_view_inlines.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_handle.c: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_handle.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_handle_inlines.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_page_directory.c: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_page_directory.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_view.c: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_view.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_view_inlines.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_view.c:
(pas_segregated_view_is_primordial_partial): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_segregated_view.h:
(pas_segregated_view_is_shared): Deleted.
(pas_segregated_view_get_shared): Deleted.
(pas_segregated_view_is_shared_handle): Deleted.
(pas_segregated_view_get_shared_handle): Deleted.
(pas_segregated_view_is_partial): Deleted.
(pas_segregated_view_get_partial): Deleted.
(pas_segregated_view_get_page_role_for_owner): Deleted.
(pas_segregated_view_get_page_role_for_allocator): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_segregated_view_kind.h:
(pas_segregated_view_kind_get_role_for_owner): Deleted.
(pas_segregated_view_kind_get_role_for_allocator): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_shared_handle_or_page_boundary.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_shared_handle_or_page_boundary_inlines.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_shared_page_directory_by_size.c: Removed.
* Source/bmalloc/libpas/src/libpas/pas_shared_page_directory_by_size.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_status_reporter.c:
(pas_status_reporter_dump_segregated_shared_page_directory): Deleted.
(dump_all_shared_page_directories_directory_callback): Deleted.
(pas_status_reporter_dump_all_shared_page_directories): Deleted.
(total_fragmentation_shared_page_directory_callback): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_utility_heap_config.c:
(pas_utility_heap_shared_page_directory_selector): Deleted.
(pas_utility_heap_config_for_each_shared_page_directory): Deleted.
(pas_utility_heap_config_dump_shared_page_directory_arg): Deleted.
* Source/bmalloc/libpas/src/test/IsoHeapPartialAndBaselineTests.cpp:
(std::forEachSharedPageDirectoryCallbackAdaptor): Deleted.
(std::forEachSharedPageDirectory): Deleted.
(std::numSharedPages): Deleted.
(std::numCommittedSharedPages): Deleted.
(std::testSimplePartialAllocations): Deleted.
(std::testFreeAroundPrimordialStop): Deleted.
(std::testFreeInterleavedAroundPrimordialStop): Deleted.
(std::testMultiplePartialsFromDifferentHeapsPerShared): Deleted.
(std::addMultiplePartialsFromDifferentHeapsPerSharedTests): Deleted.
(std::testMultiplePartialsFromDifferentThreadsPerShared): Deleted.
* Source/bmalloc/libpas/src/test/TestHarness.cpp:
(ForceExclusives::ForceExclusives): Deleted.
(ForcePartials::ForcePartials): Deleted.
* Source/bmalloc/libpas/src/test/ViewCacheTests.cpp:

Canonical link: <a href="https://commits.webkit.org/312239@main">https://commits.webkit.org/312239@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e2b74cb87dabc8ad8dfdadca9076bbc001312965

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/159319 "14 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/32747 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/25853 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/168149 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/113697 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/6f848d44-33dc-44de-84a2-da9c712bbe7d) 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/161188 "Build is in progress. Recent messages:") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/155/builds/32815 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/32734 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/123440 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/113697 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/7ef688e2-e335-4085-a44c-3a4fd5c28648) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/162276 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/155/builds/32815 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/143104 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/104107 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/b3350682-09f0-4d50-a24e-210b1fc8f843) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/155/builds/32815 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/23193 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk3-libwebrtc~~](https://ews-build.webkit.org/#/builders/173/builds/15922 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/151369 "Build is in progress. Recent messages:") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/155/builds/32815 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/20884 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/170643 "Built successfully") | | 
| [  ~~🛠 🧪 jsc-debug-arm64~~](https://ews-build.webkit.org/#/builders/171/builds/20152 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/22510 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/131643 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/153/builds/32436 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/27267 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/131755 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/35630 "Built successfully and passed tests") | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/160/builds/32380 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/142677 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/90505 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/160/builds/32380 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/170/builds/19486 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/191602 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/31891 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/49237 "Passed tests") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/157/builds/31411 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/31684 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [⏳ 🛠 watch-sim ](https://ews-build.webkit.org/#/builders/watchOS-26-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | | | 
<!--EWS-Status-Bubble-End-->

Constellation commented on Source/bmalloc/libpas/src/libpas/pas_create_basic_heap_page_caches_with_reserved_memory.c:
```diff
@@ -95,7 +94,8 @@ pas_basic_heap_page_caches* pas_create_basic_heap_page_caches_with_reserved_memo
     pas_basic_heap_page_caches* caches;
-    pas_segregated_page_config_variant segregated_variant;
+
+    /* MPTODO REMOVE PARAM ? */ (void)template_runtime_config;
```
Let's use PAS_UNUSED_PARAM

Constellation commented on Source/bmalloc/libpas/src/libpas/pas_get_page_base_and_kind_for_small_other_in_fast_megapage.h:
```diff
@@ -57,7 +57,7 @@ pas_get_page_base_and_kind_for_small_other_in_fast_megapage(uintptr_t begin, pas
     return pas_page_base_and_kind_create(
         pas_page_base_for_address_and_page_config(begin, config.small_segregated_config.base),
-        pas_small_shared_segregated_page_kind);
+        pas_small_exclusive_segregated_page_kind); /* MPTODO !!! is this the right thing to do ??? was shared */
```
Maybe we need to check?

Constellation commented on Source/bmalloc/libpas/src/libpas/pas_segregated_directory_kind.h:
```diff
@@ -34,14 +34,12 @@ PAS_BEGIN_EXTERN_C;
    describing the instantiable leaves. */
 
-enum pas_segregated_directory_kind {
+enum pas_segregated_directory_kind { /* MPTODO remove this ? */
```
Let's remove TODO comment, or let's remove this class. I think either is fine for now

Achierius committed:
[libpas] Remove partial views
https://bugs.webkit.org/show_bug.cgi?id=304655
rdar://151419388

Reviewed by Yusuke Suzuki.

Partial views have been completely disabled for some time, as the
benefits they gave (<1% memory savings) were not worth the complexity,
for reasons of security, performance, and maintanance burden.
In particular, the performance-memory tradeoff was unfavorable, and we
were able to pay for their disablement with some changes to the
object-size-class allocation layout.
As this code is unused, it should be removed to reduce complexity and
improve speed of development.  This patch tears out the implementation
for partial views, including
 - partial views
 - shared views
 - shared directories
and other affiliated structures.

* Source/bmalloc/libpas/src/libpas/jit_heap_config.c:
(jit_small_segregated_shared_page_directory_selector): Deleted.
(jit_heap_config_for_each_shared_page_directory): Deleted.
(jit_heap_config_for_each_shared_page_directory_remote): Deleted.
(jit_heap_config_dump_shared_page_directory_arg): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_all_heaps.c:
(for_each_segregated_directory_shared_page_directory_callback): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_all_shared_page_directories.c: Removed.
* Source/bmalloc/libpas/src/libpas/pas_all_shared_page_directories.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_basic_heap_page_caches.h:
(pas_basic_heap_page_caches_get_shared_page_directories): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_compact_atomic_segregated_partial_view_ptr.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_compact_segregated_shared_page_directory_ptr.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_compact_segregated_shared_view_ptr.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_enumerate_segregated_heaps.c:
(collect_shared_page_directories_shared_page_directory_callback): Deleted.
(collect_shared_page_directories_heap_callback): Deleted.
(enumerate_shared_view): Deleted.
(enumerate_partial_view): Deleted.
(shared_page_directory_view_callback): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_full_alloc_bits_inlines.h:
(pas_full_alloc_bits_create_for_view_and_directory):
(pas_full_alloc_bits_create_for_view):
(pas_full_alloc_bits_create_for_partial_but_not_primordial): Deleted.
(pas_full_alloc_bits_create_for_partial): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_heap_config_utils.c:
(pas_heap_config_utils_for_each_shared_page_directory): Deleted.
(pas_heap_config_utils_for_each_shared_page_directory_remote): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_local_allocator_config_kind.h:
(pas_local_allocator_config_kind_is_primordial_partial): Deleted.
(pas_local_allocator_config_kind_create_primordial_partial): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_local_allocator_inlines.h:
(pas_local_allocator_set_up_primordial_bump): Deleted.
(pas_local_allocator_start_allocating_in_primordial_partial_view): Deleted.
(pas_local_allocator_bless_primordial_partial_view_before_stopping): Deleted.
(pas_local_allocator_try_allocate_in_primordial_partial_view): Deleted.
(pas_local_allocator_return_memory_to_page_for_role): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_page_kind.h:
(pas_page_kind_get_segregated_role): Deleted.
(pas_page_kind_for_segregated_variant_and_role): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_segregated_page_config.h:
(pas_segregated_page_config_payload_offset_for_role): Deleted.
(pas_segregated_page_config_payload_size_for_role): Deleted.
(pas_segregated_page_config_payload_end_offset_for_role): Deleted.
(pas_segregated_page_config_enable_empty_word_eligibility_optimization_for_role): Deleted.
(pas_segregated_page_config_logging_mode_for_role): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_segregated_page_config_kind_and_role.c: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_page_config_kind_and_role.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_page_inlines.h:
* Source/bmalloc/libpas/src/libpas/pas_segregated_page_role.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_partial_view.c: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_partial_view.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_partial_view_inlines.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_handle.c: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_handle.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_handle_inlines.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_page_directory.c: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_page_directory.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_view.c: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_view.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_shared_view_inlines.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_segregated_view.c:
(pas_segregated_view_is_primordial_partial): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_segregated_view.h:
(pas_segregated_view_is_shared): Deleted.
(pas_segregated_view_get_shared): Deleted.
(pas_segregated_view_is_shared_handle): Deleted.
(pas_segregated_view_get_shared_handle): Deleted.
(pas_segregated_view_is_partial): Deleted.
(pas_segregated_view_get_partial): Deleted.
(pas_segregated_view_get_page_role_for_owner): Deleted.
(pas_segregated_view_get_page_role_for_allocator): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_segregated_view_kind.h:
(pas_segregated_view_kind_get_role_for_owner): Deleted.
(pas_segregated_view_kind_get_role_for_allocator): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_shared_handle_or_page_boundary.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_shared_handle_or_page_boundary_inlines.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_shared_page_directory_by_size.c: Removed.
* Source/bmalloc/libpas/src/libpas/pas_shared_page_directory_by_size.h: Removed.
* Source/bmalloc/libpas/src/libpas/pas_status_reporter.c:
(pas_status_reporter_dump_segregated_shared_page_directory): Deleted.
(dump_all_shared_page_directories_directory_callback): Deleted.
(pas_status_reporter_dump_all_shared_page_directories): Deleted.
(total_fragmentation_shared_page_directory_callback): Deleted.
* Source/bmalloc/libpas/src/libpas/pas_utility_heap_config.c:
(pas_utility_heap_shared_page_directory_selector): Deleted.
(pas_utility_heap_config_for_each_shared_page_directory): Deleted.
(pas_utility_heap_config_dump_shared_page_directory_arg): Deleted.
* Source/bmalloc/libpas/src/test/IsoHeapPartialAndBaselineTests.cpp:
(std::forEachSharedPageDirectoryCallbackAdaptor): Deleted.
(std::forEachSharedPageDirectory): Deleted.
(std::numSharedPages): Deleted.
(std::numCommittedSharedPages): Deleted.
(std::testSimplePartialAllocations): Deleted.
(std::testFreeAroundPrimordialStop): Deleted.
(std::testFreeInterleavedAroundPrimordialStop): Deleted.
(std::testMultiplePartialsFromDifferentHeapsPerShared): Deleted.
(std::addMultiplePartialsFromDifferentHeapsPerSharedTests): Deleted.
(std::testMultiplePartialsFromDifferentThreadsPerShared): Deleted.
* Source/bmalloc/libpas/src/test/TestHarness.cpp:
(ForceExclusives::ForceExclusives): Deleted.
(ForcePartials::ForcePartials): Deleted.
* Source/bmalloc/libpas/src/test/ViewCacheTests.cpp:

Canonical link: https://commits.webkit.org/312239@main
